source/includes/connection-snippets/scram-snippets/c-connection.c

```diff
@@ -0,0 +1,24 @@
+// begin scram connection
+#include <mongoc/mongoc.h>
+
+int
+main (int argc, char *argv[])
+{
+   mongoc_database_t *database;
+   mongoc_client_t *client;
+
+   mongoc_init ();
+
+   client = mongoc_client_new(
+      "mongodb+srv://<username>:<password>@<cluster-url>/test?retryWrites=true&w=majority"
+   );
+   database = mongoc_client_get_database (client, "test");
+
+   mongoc_database_destroy (database);
+   mongoc_client_destroy (client);
```

Comment:
Note that this program is not performing any IO and thus isn't testing that auth works. Other language examples issue a `count` command (see: [example in libmongoc docs](http://mongoc.org/libmongoc/current/tutorial.html#counting-documents)).

You may want to ensure this program is consistent with https://github.com/mongodb/docs-ecosystem/pull/587.